semver.js

```diff
@@ -799,11 +799,15 @@ Range.prototype.intersects = function (range, options) {
 
   return this.set.some(function (thisComparators) {
     return range.set.some(function (rangeComparators) {
-      return isSatisfiable(thisComparators, options) && thisComparators.every(function (thisComparator) {
-        return isSatisfiable(rangeComparators, options) && rangeComparators.every(function (rangeComparator) {
-          return thisComparator.intersects(rangeComparator, options)
+      return (
+        isSatisfiable(thisComparators, options) &&
```

Comment:
This could also be moved up a line.  No sense walking `range.set` if `thisComparators` isn't satisfiable.